File: android/app/src/main/java/com/sanyi_client/BundleManagerModule.java
package com.sanyi_client;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

public class BundleManagerModule extends ReactContextBaseJavaModule {
    BundleManagerModule(ReactApplicationContext context) {
        super(context);
    }

    @NonNull
    @Override
    public String getName() {
        return "BundleManagerModule";
    }

    @ReactMethod
    public void loadRNBundle(String path, Promise promise) {
        getReactApplicationContext().getCatalystInstance().loadSplitBundleFromFile(path, path);
        promise.resolve(true);
    }
}

File: android/app/src/main/java/com/sanyi_client/BundleManagerModule.kt
package com.sanyi_client

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class BundleManagerModule internal constructor(context: ReactApplicationContext?) :
    ReactContextBaseJavaModule(context) {
    override fun getName(): String {
        return "BundleManagerModule"
    }

    @ReactMethod
    fun loadRNBundle(path: String?, promise: Promise) {
        reactApplicationContext.catalystInstance.loadSplitBundleFromFile(path, path)
        promise.resolve(true)
    }
}